san/impl/params_resolver.cpp

```diff
@@ -38,7 +36,7 @@ void ParamsValidator::visit(RealType*) {}
 
 void ParamsValidator::visit(BooleanType*) {}
 
-void ParamsValidator::visit(ArrayType* node) { node->length->accept(*this); }
+void ParamsValidator::visit(ArrayType*) {}
```

Comment:
Even though the length has to be a constant (so not a routine call), it can still be an array of records, which can have nested variable declarations that can be initialized with a routine call